x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -3122,6 +3122,65 @@ public void testSpatialTypesAndStatsUseDocValues() {
         var from = as(agg.child(), EsRelation.class);
     }
 
+    /**
+     * Expects
+     * Limit[1000[INTEGER]]
+     * \_Aggregate[[],[SPATIALCENTROID(location{f}#9) AS centroid]]
+     *   \_EsRelation[airports][abbrev{f}#5, location{f}#9, name{f}#6, scalerank{f}..]
+     */
+    public void testSpatialTypesAndStatsUseDocValuesWithEval() {
+        var plan = planAirports("""
+            from test
+            | stats centroid = st_centroid_agg(to_geopoint(location))
+            """);
+
+        var limit = as(plan, Limit.class);
+        var agg = as(limit.child(), Aggregate.class);
+        assertThat(Expressions.names(agg.aggregates()), contains("centroid"));
+        assertTrue("Expected GEO_POINT aggregation for STATS", agg.aggregates().stream().allMatch(aggExp -> {
+            var alias = as(aggExp, Alias.class);
+            var aggFunc = as(alias.child(), AggregateFunction.class);
+            var aggField = as(aggFunc.field(), FieldAttribute.class);
+            return aggField.dataType() == GEO_POINT;
+        }));
+
+        as(agg.child(), EsRelation.class);
+    }
+
+    /**
+     * Expects:
+     * Eval[[first_name{f}#6 AS first_name]]
+     * \_Limit[1000[INTEGER]]
+     *   \_EsRelation[test][_meta_field{f}#11, emp_no{f}#5, first_name{f}#6, ge..]
+     * NOTE: The convert function to_string is removed, since the types match
+     */
+    public void testToStringOnStringWrittenAway() {
+        var plan = plan("from test | eval first_name = to_string(first_name)");
+        var eval = as(plan, Eval.class);
+        var alias = as(eval.fields().get(0), Alias.class);
+        var fa = as(alias.child(), FieldAttribute.class);
+        assertThat(fa.name(), equalTo("first_name"));
+        var limit = as(eval.child(), Limit.class);
+        as(limit.child(), EsRelation.class);
+    }
+
+    /**
+     * Expects:
+     * Eval[[emp_no{f}#5 AS emp_no]]
+     * \_Limit[1000[INTEGER]]
+     *   \_EsRelation[test][_meta_field{f}#11, emp_no{f}#5, first_name{f}#6, ge..]
+     * NOTE: The convert function to_integer is removed, since the types match
+     */
+    public void testToIntegerOnIntegerWrittenAway() {
+        var plan = plan("from test | eval emp_no = to_integer(emp_no)");
+        var eval = as(plan, Eval.class);
+        var alias = as(eval.fields().get(0), Alias.class);
+        var fa = as(alias.child(), FieldAttribute.class);
+        assertThat(fa.name(), equalTo("emp_no"));
+        var limit = as(eval.child(), Limit.class);
+        as(limit.child(), EsRelation.class);
```

Comment:
I think these two tests could be refactored into a single, parameterized test that goes over all relevant data types.

Reply:
Yes, but the `mapping-basic.json` mapping used for this entire class is lacking in type diversity to achieve that. Perhaps I should look around for other unit tests that do not do a complete index mapping for such coverage.

Reply:
I realize I can just add a bunch of new fields to the mapping-basic.json, so I did that.

Reply:
Well, that caused a lot of other tests to fail, due to the extra and unexpected fields, so I've made a new mapping file with these fields, used just in that test.

Reply:
There is `mapping-all-types.json` that I used in the past for cases like this.